### PR TITLE
xp globe: add skill icon scaling

### LINKED ISSUE #10116: Add scaling to the skill icons in XP Orbs
Currently, when scaling an XP orb below the default size of 35, the edges of the skill can clip out of the edges of the orb, which looks very bad. Conversely, making the orb larger than 35 results in a lot of empty space around the icon, which looks bad.

To fix this, I propose that the icon scale. Scaling smaller than default will have to cap somewhere, however. The user should be given an option as to whether they want the icon to scale or not. It might also be a good idea to add a lower cap to the size of the orbs, now that we can easily do this.

### PR DESCRIPTION
**Pull request by NathanielPather**
Closes #10116 

### Features
Skill icons scales with xpOrbSize.
xpOrbSize changed to have minimum value of 20.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java [hunk 1]
```diff
@@ -221,11 +223,13 @@ private void drawSkillImage(Graphics2D graphics, XpGlobe xpGlobe, int x, int y)
...
+
 		graphics.drawImage(
-			skillImage,
-			x + (config.xpOrbSize() / 2) - (skillImage.getWidth() / 2),
-			y + (config.xpOrbSize() / 2) - (skillImage.getHeight() / 2),
-			null
+				resized,
+				x + (config.xpOrbSize() / 2) - (resized.getWidth() / 2),
+				y + (config.xpOrbSize() / 2) - (resized.getHeight() / 2),
+				null
```
Comment (Hydrox6) [inline runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java]: Remove the excess indents

File: runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesConfig.java [hunk 2]
```diff
@@ -146,6 +146,9 @@ default int progressArcStrokeWidth()
 		return 2;
 	}
 
+	@Range (
+			min = 20
```
Comment (Hydrox6) [inline runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesConfig.java]: Remove the excess indent

File: runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java [hunk 3]
```diff
@@ -53,6 +53,8 @@
 import net.runelite.client.ui.overlay.OverlayUtil;
 import net.runelite.client.ui.overlay.components.LineComponent;
 import net.runelite.client.ui.overlay.components.PanelComponent;
+import net.runelite.client.util.ImageUtil;
+
```
Comment (Hydrox6) [inline runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java]: Remove the extra newline

Comment (NathanielPather) [reply runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java]: No idea how I added the extra indents. Thanks tho.

File: runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java [hunk 4]
```diff
@@ -221,10 +222,12 @@ private void drawSkillImage(Graphics2D graphics, XpGlobe xpGlobe, int x, int y)
 			return;
 		}
 
+		BufferedImage resized = ImageUtil.resizeImage(skillImage, config.xpOrbSize() / 2, config.xpOrbSize() / 2);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java]: I don't think this should be resizing the image every single render cycle

Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java]: To expand on this, you can generate the image once in the plugin, and update it on config change accordingly, then get that image from the overlay so it is not generated once per frame.

**Comment by Hydrox6** [comment]
This was added in e7f50433a352ab996b7b5b6114b504194312000d

**Pull request closed without merging**